R/data-set.R

```diff
@@ -0,0 +1,141 @@
+#' @title DataSet
+#' @docType class
+#' @description  A wrapper around DataRepository exposing convenience methods to use and manipulate dataSets
+#' (typically observed data) containing an X column, a Y column and potentially an Error columns
+#' @format NULL
+DataSet <- R6::R6Class(
+  "DataSet",
+  inherit = Printable,
+  cloneable = FALSE,
+  active = list(
+    #' @field name The name of the DataSet
+    name = function(value) {
+      if (missing(value)) {
+        return(self$dataRepository$name)
+      }
+      self$dataRepository$name <- value
+    },
+    #' @field xDimension Dimension in which the xValues are defined
+    xDimension = function(value) {
+      if (missing(value)) {
+        return(private$.xValues$dimension)
+      }
+      private$setColumnDimension(private$.xValues, value)

```

Comment:
I have define a method for each `set`a ction so that we can call it for x, y and later error